### DISCUSSION TITLE
Range Rate vs Velocity Vector

### DISCUSSION THREAD

**Discussion by blairwheadon** [skyfielders/python-skyfield#748] [Q&A] [answered]
OK this question isn't 100% Skyfield related, but I'm guessing the smart people here may know the answer.

I'm doing a project on planetary speed and am trying to shift over to using Skyfield instead of flat file extracts from JPL Horizons.

JPL has fields called '[Target Range and Range-Rate](https://ssd.jpl.nasa.gov/horizons/manual.html)'

defined as:

_Target apparent range ("delta", light-time aberrated) and range-rate ("delta-dot") relative to the observer. A positive "deldot" means the target is moving away from the observer, negative indicates movement toward the observer._

Range-rate values are totally different from Skyfield speed() values

In Horizons, speed is abbreviated as vMagOb and is defined as:

_Magnitude of the targets' velocity with respect to the Sun center and the observer at the time light left the target to be observed (print time minus down-leg light-time). These are absolute values of the velocity vectors (total speeds) and do NOT indicate direction of motion._

These values match the Skyfield speed() values.

My question: What's the difference between range-rate and speed? Other than the +/- sign used by range-rate, it's not clear to me what's behind the different results. Can 'apparent' range-rate really account for the big differences I'm seeing?

For example: May 26 2012 15:00 UTC: Mercury geocentric Deldot = 0.2753635, velocity magnitude relative to observer = 87.9038400. Both are in km/s. Values from Horizons.

Thanks in advance.

**Comment by blairwheadon** [comment, marked as answer]
I figured it out. range-rate (deldot) is the first derivative of distance between the bodies.

This is different the magnitude of the velocity vector of the target body. Range-rate takes into consideration the speed of the observer as well, whereas magnitude of the velocity vector wouldn't.

**Comment by davidmikolas** [comment]
Yes it seems you've got it just right. All velocities (and speeds) are relative to something. I'm pretty sure that the Skyfield velocity and speed are going to be relative to the solar system barycenter which is close to and often but not always inside the Sun. You might think its heliocentric (with respect to the Sun itself) but no, because Jupiter, Saturn and Uranus move the Sun around so much. 

Range-rate is discussed further in [What are the differences (if any) between the techniques of range-rate and delay-doppler measurements?](https://space.stackexchange.com/q/55199/12102) It's the rate of change of range and important because this is related to (but not the same as) what round-trip Doppler measurements from deep space spacecraft measure. The subtleties are numerous; round trip times can be hours or a day and there are relativistic corrections due to both speed and the gravitational well of the Sun.

Anyway, you solved it - you're getting *barycentric* velocity and speed in Skyfield, and something related to Doppler shift in Horizons.
